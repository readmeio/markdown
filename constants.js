export const MERMAID_BLOCK_LANG = 'mermaid-render-svg';
