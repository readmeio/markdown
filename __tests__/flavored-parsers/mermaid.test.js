import { MERMAID_BLOCK_LANG } from '../../constants';
import { mdast } from '../../index';

describe('Parse Mermaid blocks', () => {
  it('parses a mermaid block', () => {
    const doc = `
~~~${MERMAID_BLOCK_LANG}
graph LR
    A[Will it blend?] --> B[Yes]
~~~
`;

    expect(mdast(doc)).toMatchSnapshot();
  });
});
