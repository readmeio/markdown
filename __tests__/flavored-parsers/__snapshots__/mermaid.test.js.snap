// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse Mermaid blocks parses a mermaid block 1`] = `
Object {
  "children": Array [
    Object {
      "data": Object {
        "hName": "mermaid",
        "hProperties": Object {
          "value": "graph LR
    A[Will it blend?] -> B[Yes]",
        },
      },
      "type": "mermaid",
      "value": "graph LR
    A[Will it blend?] -> B[Yes]",
    },
  ],
  "type": "root",
}
`;
